src/main/java/com/google/sps/DataServlet.java

```diff
@@ -169,16 +161,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     truncatedGraph = currDataGraph.getReachableNodes(nodeNameParam, depthNumber);
 
-    // We set the headers in the following 3 scenarios:
+    if (nodeNameParam.length() == 0 && truncatedGraph.equals(currDataGraph.graph())) {
+      // If we are not filtering the graph or limiting its depth, show all mutations of all nodes
+      filteredMutationIndices = defaultIndices;
+    } else {
+      // Get the names of all the displayed nodes and find all indices of mutations
+      // that mutate any of them
+      Set<String> truncatedGraphNodeNames = Utility.getNodeNamesInGraph(truncatedGraph);
+      // Also get mutations relevant to the searched node if it is not an empty string
+      if (nodeNameParam.length() != 0) {
+        truncatedGraphNodeNames.add(nodeNameParam);
+      }
+      filteredMutationIndices =
+          Utility.findRelevantMutations(truncatedGraphNodeNames, mutationIndicesMap, mutList);
```

Comment:
similar to what we discussed during the team meeting today, here the `truncatedGraphNodeNames` should be the ones without lines 172-174. If the `nodeNameParam` is deleted, you only want to show the indices relevant to the rest of the graph. 
The one on line 179 should be the nodeNames including `nodeNameParam`. Seems like in this case swapping this line with 172-174 would work, but also feel free to integrate this with Cindy's in a later PR.

Reply:
Since it's not that big a change I made it here :)

Reply:
Actually, what if the searched node doesn't exist yet but is added in the future. Then if we changed the logic we would never be able to detect it.

Reply:
I think @haoyxc addressed this in her PR, so resolving this for now!